### DISCUSSION TITLE
How to setup lsp configuration for solidity and vue in lazyvim?

### DISCUSSION THREAD

**Discussion by Skythrill256** [LazyVim/LazyVim#2111] [General]
I am confused

**Comment by dpetka2001** [comment]
What exactly confuses you? Did you read the [LSP](https://www.lazyvim.org/plugins/lsp#nvim-lspconfig) docs to see how you can set it up? There's also the [example](https://www.lazyvim.org/configuration/examples) file which shows how you can set up `pyright` and `tsserver`. Other than that you should ensure that you have configured correctly your LSP `root_dir` by looking at [lspconfig](https://github.com/neovim/nvim-lspconfig/tree/master/lua/lspconfig/server_configurations) docs and see how the server you want to use expects the `root_dir` to be set (either via some file pattern like `package.json`, `.git` or something else).
